source/gameengine/Ketsji/KX_FontObject.cpp

```diff
@@ -113,6 +117,85 @@ void KX_FontObject::ProcessReplica()
 	KX_GameObject::ProcessReplica();
 }
 
+void KX_FontObject::AddMeshUser()
+{
+	m_meshUser = new RAS_TextUser(m_pClient_info);
+
+	// set the part of the mesh slot that never change
+	float *fl = GetOpenGLMatrixPtr()->getPointer();
+	m_meshUser->SetMatrix(fl);
+
+	RAS_BucketManager *bucketManager = GetScene()->GetBucketManager();
+	bool created = false;
+	RAS_MaterialBucket *bucket = bucketManager->FindBucket(GetTextMaterial(), created);
+
+	// If the material bucket is just created then we add a new mesh slot.
+	if (created) {
+		bucket->AddMesh(NULL);
+	}
+
+	/* We copy the original mesh slot which is at the begin of the list, if it's not the case it
+	 * doesn't matter as the mesh slot are all similar exepted their mesh user pointer which is
+	 * set to NULL in copy. By copying instead of adding a mesh slot we reuse the same display
+	 * array bucket.
+	 */
+	RAS_MeshSlot *ms = bucket->CopyMesh(*bucket->msBegin());
+	ms->SetMeshUser(m_meshUser);
+	ms->SetDeformer(NULL);
+	m_meshUser->AddMeshSlot(ms);
+}
+
+void KX_FontObject::UpdateBuckets()
+{
+	// Update datas and add mesh slot to be rendered only if the object is not culled.
+	if (m_bVisible && m_meshUser) {
+		if (m_pSGNode->IsDirty()) {
+			GetOpenGLMatrix();
+		}
+
+		// Allow for some logic brick control
+		if (GetProperty("Text")) {
+			m_text = split_string(GetProperty("Text")->GetText());
+		}
+
+		// update the animated color
+		GetObjectColor().getValue(m_color);
```

Comment:
The issue with text object is that they use multitexturing to use a font texture without material. I'm fully agree with you that a material would be the nicest case, but currently it's pretty complicated as we have to implicitly create a specific material for mesh which is able to use a font texture and that is not reused by other meshes. This is a similar case to how we manage GI materials and i have a lot of issues with it.

@youle31 : I will investigate deeper on the question, but i think that in any case we will need a fake material as KX_TextMaterial to make compatibility with shadeless text as before.